homeassistant/components/intent/timers.py

```diff
@@ -0,0 +1,816 @@
+
+
+class TimerNotFoundError(intent.IntentHandleError):
+    """Error when a timer could not be found by name or start time."""
+
+    def __init__(self) -> None:
+        """Initialize error."""
+        super().__init__("Timer not found", TIMER_NOT_FOUND_RESPONSE)
+
+
+class MultipleTimersMatchedError(intent.IntentHandleError):
+    """Error when multiple timers matched name or start time."""
+
+    def __init__(self) -> None:
+        """Initialize error."""
+        super().__init__("Multiple timers matched", MULTIPLE_TIMERS_MATCHED_RESPONSE)
+
+
+class TimerManager:
+    """Manager for intent timers."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize timer manager."""
+        self.hass = hass
+
+        # timer id -> timer
+        self.timers: dict[str, TimerInfo] = {}
+        self.timer_tasks: dict[str, asyncio.Task] = {}
+
+        self.handlers: list[TimerHandler] = []
+
+    def register_handler(self, handler: TimerHandler) -> Callable[[], None]:
+        """Register a timer handler.
+
+        Returns a callable to unregister.
+        """
+        self.handlers.append(handler)
+        return lambda: self.handlers.remove(handler)
+
+    async def start_timer(
+        self,
+        hours: int | None,
+        minutes: int | None,
+        seconds: int | None,
+        language: str,
+        device_id: str | None,
+        name: str | None = None,
+    ) -> str:
+        """Start a timer."""
+        total_seconds = 0
+        if hours is not None:
+            total_seconds += 60 * 60 * hours
+
+        if minutes is not None:
+            total_seconds += 60 * minutes
+
+        if seconds is not None:
+            total_seconds += seconds
+
+        timer_id = ulid.ulid_now()
+        created_at = time.monotonic_ns()
+        timer = TimerInfo(
+            id=timer_id,
+            name=name,
+            start_hours=hours,
+            start_minutes=minutes,
+            start_seconds=seconds,
+            seconds=total_seconds,
+            language=language,
+            device_id=device_id,
+            created_at=created_at,
+            updated_at=created_at,
+        )
+
+        # Fill in area/floor info
+        device_registry = dr.async_get(self.hass)
+        if device_id and (device := device_registry.async_get(device_id)):
+            timer.area_id = device.area_id
+            area_registry = ar.async_get(self.hass)
+            if device.area_id and (
+                area := area_registry.async_get_area(device.area_id)
+            ):
+                timer.floor_id = area.floor_id
+
+        self.timers[timer_id] = timer
+        self.timer_tasks[timer_id] = self.hass.async_create_background_task(
+            self._wait_for_timer(timer_id, total_seconds, created_at),
+            name=f"Timer {timer_id}",
+        )
+
+        await asyncio.gather(
+            *(handler(TimerEventType.STARTED, timer) for handler in self.handlers)
+        )
```

Comment:
I think that we should make these callbacks, since they are blocking the returning of the timer. If a handler wants to do something async, they can schedule that on the event loop themselves.